relays/bin-substrate/src/cli/init_bridge.rs

```diff
@@ -108,10 +108,9 @@ impl InitBridge {
 	/// Run the command.
 	pub async fn run(self) -> anyhow::Result<()> {
 		select_bridge!(self.bridge, {
-			let source_client = crate::rialto_millau::source_chain_client::<Source>(self.source).await?;
-			let target_client = crate::rialto_millau::target_chain_client::<Target>(self.target).await?;
-			let target_sign =
-				Target::target_signing_params(self.target_sign).map_err(|e| anyhow::format_err!("{}", e))?;
+			let source_client = self.source.into_client::<Source>().await?;
+			let target_client = self.target.into_client::<Target>().await?;
+			let target_sign = self.target_sign.into_keypair::<Target>()?;
```

Comment:
Slightly off-topic, but how does the Rust compiler know to resolve macros in the correct order?

For example, the `into_client()` and `into_keypair()` methods only exist after the `declare_chain_options!` macro has been evaluated. How does the compiler know to evaluate `declare_chain_options!` first and only after that code's been generated evaluate `select_bridge!`?

Reply:
Macros are expanded before the compilation, so the code is generated first and only then compiled, so it's not really an issue here.

The interesting part is in what order macros are declared and resolved, and interestingly it's simply top-down in the order they are listed. (i.e. `macro_rules` need to be before usage).

Reply:
> Macros are expanded before the compilation, so the code is generated first and only then compiled, so it's not really an issue here.

Actually yeah, that makes a lot of sense

> The interesting part is in what order macros are declared and resolved, and interestingly it's simply top-down in the order they are listed. (i.e. macro_rules need to be before usage).

This is kinda dumb, but given the first thing you said also makes sense